test/functional/commands/geo-location-e2e-specs.js

```diff
@@ -0,0 +1,53 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { retryInterval } from 'asyncbox';
+import { AndroidDriver } from '../../../..';
+import sampleApps from 'sample-apps';
+
+
+chai.should();
+chai.use(chaiAsPromised);
+
+let driver;
+let caps = {
+  app: sampleApps('gpsDemo-debug'),
+  deviceName: 'Android',
+  platformName: 'Android'
+};
+
+describe("geo-location", function () {
+  before(async () => {
+    driver = new AndroidDriver();
+    await driver.createSession(caps);
+  });
+  after(async () => {
+    await driver.deleteSession();
+  });
+
+  it('should set geo location @skip-ci', async () => {
+    let getText = async () => {
+      let els = await driver.findElOrEls('class name', 'android.widget.TextView', true);
+      return await driver.getText(els[1].ELEMENT);
+    };
+
+    let latitude = '27.17';
+    let longitude = '78.04';
+
+    let text = await getText();
+    text.should.not.include(`Latitude: ${latitude}`);
+    text.should.not.include(`Longitude: ${longitude}`);
+
+    await driver.setGeoLocation({latitude, longitude});
+
+    // wait for the text to change
+    await retryInterval(3, 1000, async () => {
```

Comment:
The failure in CI seems to be because we don't retry this enough times. Try bumping from `3` to `6` and rerunning. If it still fails, we can regroup.